secboot/secboot_sb.go

```diff
@@ -102,7 +102,16 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 		return res, nil
 	}
 
-	mapperName := name + "-" + randutilRandomKernelUUID()
+	uuid, uuidErr := randutilRandomKernelUUID()
+	if uuidErr != nil {
+		// Duplicate the exact behaviour of randutil.RandomKernelUUID()
+		// before the change to return an error. Callers of this function
+		// make assumptions on what could have caused an error, so adding
+		// a new error source would require a rework of sensitive code.
+		panic("cannot read kernel generated uuid")
```

Comment:
Unless @pedronis feels differently I think it's fine to return unlockRes, err here now? The result should effectively be the same as the unlock code will exit and the device will be unusable. But to me having an error return here feels more consistent with the rest of the code. Having said this, **if** we panic I think we want to include the error from the  `randutilRandomKernelUUID` so something like `panic(fmt.Sprintf("cannot read kernel generated uuid: %v", err))`

Reply:
Thanks!

Reply:
Fixed both cases and added unit tests.